bundles/org.openhab.binding.dwdpollenflug/src/main/java/org/openhab/binding/dwdpollenflug/internal/handler/DWDPollenflugBridgeHandler.java

```diff
@@ -0,0 +1,213 @@
+        CompletableFuture<@Nullable DWDPollenflug> f = new CompletableFuture<>();
+        Request request = client.newRequest(URI.create(DWD_URL));
+
+        request.method(HttpMethod.GET).timeout(2000, TimeUnit.SECONDS).send(new BufferingResponseListener() {
+            @NonNullByDefault({})
+            @Override
+            public void onComplete(Result result) {
+                final HttpResponse response = (HttpResponse) result.getResponse();
+                if (result.getFailure() != null) {
+                    Throwable e = result.getFailure();
+                    if (e instanceof SocketTimeoutException || e instanceof TimeoutException) {
+                        f.completeExceptionally(new DWDPollingException("Request timeout", e));
+                    } else {
+                        f.completeExceptionally(new DWDPollingException("Request failed", e));
+                    }
+                } else if (response.getStatus() != 200) {
+                    f.completeExceptionally(new DWDPollingException(getContentAsString()));
+                } else {
+                    try {
+                        DWDPollenflug pollenflugJSON = gson.fromJson(getContentAsString(), DWDPollenflug.class);
+                        f.complete(pollenflugJSON);
+                    } catch (JsonSyntaxException ex2) {
+                        f.completeExceptionally(new DWDPollingException("Parsing of response failed"));
+                    }
+                }
+            }
+        });
+
+        return f;
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof DWDPollenflugRegionHandler) {
+            logger.debug("Register region listener.");
+            if (regionListeners.add((DWDPollenflugRegionHandler) childHandler)) {
+                startPolling();
+            } else {
+                logger.warn("Tried to add listener {} but it was already present. This is probably an error.",
+                        childHandler);
+            }
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof DWDPollenflugRegionHandler) {
+            logger.debug("Unregister region listener.");
+            if (!regionListeners.remove((DWDPollenflugRegionHandler) childHandler)) {
+                logger.warn("Tried to remove listener {} but it was not registered. This is probably an error.",
+                        childHandler);
+            }
+        }
+    }
+
+    public void notifyOnUpdate(@Nullable DWDPollenflug newState) {
+        if (newState != null) {
+            pollenflug = newState;
+            updateProperties(newState.getProperties());
+            regionListeners.forEach(listener -> listener.notifyOnUpdate(newState));
+            newState.getChannelsStateMap().forEach(this::updateState);
+        }
+    }

```

Comment:
```suggestion
    public void notifyOnUpdate(@Nullable DWDPollenflug pollenflug) {
        if (pollenflug != null) {
            this.pollenflug = pollenflug;
            updateProperties(newState.getProperties());
            regionListeners.forEach(listener -> listener.notifyOnUpdate(pollenflug));
            pollenflug.getChannelsStateMap().forEach(this::updateState);
        }
    }

```

a `State` is something different in openHAB, this may result in confusion.

Reply:
Renamed